src/docs/release/breaking-changes/route-transition-record-and-transition-delegate.md

```diff
@@ -0,0 +1,139 @@
+  @override
+  Iterable<RouteTransitionRecord> resolve({
+    List<RouteTransitionRecord> newPageRouteHistory,
+    Map<RouteTransitionRecord, RouteTransitionRecord> locationToExitingPageRoute,
+    Map<RouteTransitionRecord, List<RouteTransitionRecord>> pageRouteToPagelessRoutes,
+  }) {
+    final List<RouteTransitionRecord> results = <RouteTransitionRecord>[];
+
+    for (final RouteTransitionRecord pageRoute in newPageRouteHistory) {
+      if (pageRoute.isEntering) {
+        pageRoute.markForAdd();
+      }
+      results.add(pageRoute);
+
+    }
+    for (final RouteTransitionRecord exitingPageRoute in locationToExitingPageRoute.values) {
+      exitingPageRoute.markForRemove();
+      final List<RouteTransitionRecord> pagelessRoutes = pageRouteToPagelessRoutes[exitingPageRoute];
+      if (pagelessRoutes != null) {
+        for (final RouteTransitionRecord pagelessRoute in pagelessRoutes) {
+          pagelessRoute.markForRemove();
+        }
+      }
+      results.add(exitingPageRoute);
+
+    }
+    return results;
+  }
+}
+```
+
+Code after migration:
+
+```dart
+import 'package:flutter/widgets.dart';
+
+class NoAnimationTransitionDelegate extends TransitionDelegate<void> {
+  @override
+  Iterable<RouteTransitionRecord> resolve({
+    List<RouteTransitionRecord> newPageRouteHistory,
+    Map<RouteTransitionRecord, RouteTransitionRecord> locationToExitingPageRoute,
+    Map<RouteTransitionRecord, List<RouteTransitionRecord>> pageRouteToPagelessRoutes,
+  }) {
+    final List<RouteTransitionRecord> results = <RouteTransitionRecord>[];
+
+    for (final RouteTransitionRecord pageRoute in newPageRouteHistory) {
+      // Renames isEntering to isWaitingForEnteringDecision.
+      if (pageRoute.isWaitingForEnteringDecision) {
+        pageRoute.markForAdd();
+      }
+      results.add(pageRoute);
+
+    }
+    for (final RouteTransitionRecord exitingPageRoute in locationToExitingPageRoute.values) {
+      // Checks the isWaitingForExitingDecision before calling the markFor methods.
+      if (exitingPageRoute.isWaitingForExitingDecision) {
+        exitingPageRoute.markForRemove();
+        final List<RouteTransitionRecord> pagelessRoutes = pageRouteToPagelessRoutes[exitingPageRoute];
+        if (pagelessRoutes != null) {
+          for (final RouteTransitionRecord pagelessRoute in pagelessRoutes) {
+            pagelessRoute.markForRemove();
+          }
+        }
+      }
+      results.add(exitingPageRoute);
+
+    }
+    return results;
+  }
+}
+```
+
+## Timeline
+
+This change landed in v1.18.0.
+
+## References
+
+API documentation:
+* [`Navigator`][]
+
+Relevant issue:
+* [Issue 45938: Navigator 2.0][]
+
+Relevant PR:
+* [PR 55998: Fixes the navigator pages update crashes when there is still route waiting for popping animation to finish][]
+
+
+[Issue 45938: Navigator 2.0]: {{site.github}}/flutter/flutter/issues/45938
+[`Navigator`]: {{site.api}}/flutter/widgets/Navigator-class.html
```

Comment:
Does the Navigator page on api.flutter.dev describe the new API? If not, please use the master API link.